# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.h
# Commit: 6c129bd72b61
# Full Hash: 6c129bd72b61cf6d096dcec6c7bb188e3ddb37c7
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r 4b0a4fcc6894 -r 6c129bd72b61 dom/workers/ScriptLoader.h
--- a/dom/workers/ScriptLoader.h	Wed Dec 14 14:55:34 2022 +0000
+++ b/dom/workers/ScriptLoader.h	Wed Dec 14 14:55:35 2022 +0000
@@ -12,6 +12,7 @@
 #include "mozilla/dom/WorkerCommon.h"
 #include "mozilla/dom/WorkerLoadContext.h"
 #include "mozilla/dom/WorkerRef.h"
+#include "mozilla/dom/workerinternals/WorkerModuleLoader.h"
 #include "mozilla/Maybe.h"
 #include "nsIContentPolicy.h"
 #include "nsStringFwd.h"
@@ -232,6 +233,8 @@
     return NS_OK;
   }
 
+  void InitModuleLoader();
+
   void TryShutdown();
 
   nsTArray<RefPtr<ThreadSafeRequestHandle>> GetLoadingList();